### PR TITLE
[server] add vscode(-insiders) ouath2 clients

### PR DESCRIPTION
**Pull request by filiptronicek**
## Description
It adds 2 new oauth2 clients for Gitpod Server: VS Code Desktop and VS Code Desktop Insiders.

## Related Issue(s)
#3733

## How to test

Check that the local app is still working.

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
@filiptronicek: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

File: components/proxy/conf/Caddyfile
Comment (geropl) [inline components/proxy/conf/Caddyfile:213]: @filiptronicek Could you elaborate on the motivation to allow accessing this endpoint from all domains? :thinking:

File: components/server/src/oauth-server/db.ts
Comment (geropl) [reply components/server/src/oauth-server/db.ts:48]: @csweichel Ping

File: components/gitpod-db/src/typeorm/user-db-impl.ts
Comment (geropl) [inline components/gitpod-db/src/typeorm/user-db-impl.ts:422]: nit: we never use `String`. In this case maybe `.toString()` would be a more idiomatic solution.

File: components/proxy/conf/Caddyfile
Comment (akosyakov) [reply components/proxy/conf/Caddyfile:213]: because VS Code Desktop does not provide origin, but only bearer auth

Comment (geropl) [reply components/proxy/conf/Caddyfile:213]: Thx for the explanation, make sense!

File: components/server/src/oauth-server/db.ts
Comment (geropl) [reply components/server/src/oauth-server/db.ts:48]: /hold

IMO the set is absolutely fine. /holding here in case you want more verification @akosyakov

**Review by geropl — APPROVED** [review]
Code LGTM, tested and local VSCode (still) works :+1:

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: cef5d445e51d2b1097e454f8ff664fad74311616</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/6272#pullrequestreview-794098911" title="Approved">geropl</a>*

Associated issue: *#3733*

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/gitpod-io/gitpod/blob/main/OWNERS)~~ [geropl]
- ~~[components/gitpod-db/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-db/OWNERS)~~ [geropl]
- ~~[components/proxy/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/proxy/OWNERS)~~ [geropl]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [geropl]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/server/src/oauth-server/db.ts
Comment (akosyakov) [reply components/server/src/oauth-server/db.ts:48]: I actually recalled now that I wanted to move app specification in ide config that changing scopes does not require meta deployment.

**Comment by akosyakov** [comment]
Added team IDE because merging actually blocked by me 😅

**Comment by jeanp413** [comment]
@geropl could you review this :pray:

**Review by geropl — APPROVED** [review]
LGTM

**Pull request merged**
